### PR TITLE
B-17965 Send Email - Gov't Counselor Submission (Before TOO Approval)

### PR DESCRIPTION
**Pull request by WeatherfordAaron**
## Summary

The automated test that I implemented just confirms that the correct email body is presented.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
1. In terminal run ```EMAIL_BACKEND=ses AWS_REGION=us-gov-west-1  aws-vault exec transcom-gov-dev -- make server_run```
2. On a separate terminal tab run ```make client_run```
3. As a standard user, create a move. During the move creation process, make sure you enter the email address as something you have access to check.
4. You should receive an initial email for creating the move.
5. Log into the Office side as a Services Counselor.
6. Find the move and update the Orders section per standard procedure.
7. After the Orders section has been updated, click the Submit move button.
8. You should receive a confirmation about the change in status for your move. 

### Screenshots
Screenshot 2024-01-17 at 12 09 32 PM

### CODE REVIEW

**Review by taeJungCaci — COMMENTED** [review]
In the `move_task_order_test.go`, add 		`handlerConfig.SetNotificationSender(notifications.NewStubNotificationSender("milmovelocal"))`
to line 260 below `handlerConfig := suite.HandlerConfig()`

This should fix the null server_test issue.

image

File: pkg/handlers/ghcapi/move_task_order.go [hunk 1]
```diff
@@ -221,6 +224,13 @@ func (h UpdateMTOStatusServiceCounselingCompletedHandlerFunc) Handle(params move
 		})
 }
 
+func (h UpdateMTOStatusServiceCounselingCompletedHandlerFunc) SendMoveCounseledEmail(appCtx appcontext.AppContext, moveTaskOrderID uuid.UUID) {
```
Comment (taeJungCaci) [inline pkg/handlers/ghcapi/move_task_order.go:227]: I think you can remove this func and move it directly into the line 202.

Comment (WeatherfordAaron) [reply pkg/handlers/ghcapi/move_task_order.go]: updated it and pushed just now

**Review by KonstanceH — APPROVED** [review]

**Review by taeJungCaci — APPROVED** [review]
It might be just my outlook, but there are no margins between the bullet points and under "What this means to you:".

Code looks good.

image

**Comment by JamesHawks224** [comment]
The following appear to be boilerplate:
* Verification Steps for the Author
* Verification Steps for Reviewers
* Setup to Run the Code
* Frontend
* Backend
* Database

If they aren't adding anything significantly useful, they're cluttering the PR. 
Suggest removing them.

**Pull request merged**